lib/daru/io/base.rb

```diff
@@ -1,15 +1,21 @@
+require 'daru'
+
 module Daru
   module IO
     class Base
-      def initialize(binding)
-        args = method(__method__).parameters.map do |_, name|
-          [name, binding.local_variable_get(name.to_s)]
-        end.to_h
-
-        args.each do |k, v|
-          instance_variable_set("@#{k}", v)
-          define_singleton_method(k) { instance_variable_get("@#{k}") }
-        end
+      def optional_gem(dependency, version=nil, requires: nil)
+        gem dependency, version
+        require requires || dependency
+      rescue LoadError
+        statement =
```

Comment:
I believe it is good enough, only thing I'd propose (maybe later) to add the name or description of exact importer or exporter requiring the gem.